[spdlog] build failure with benchmark feature

**Host Environment**
- OS: [Microsoft Windows [Version 10.0.19042.572]]
- Compiler: Visual Studio  Version: 16.7.6  VS Tools Version: 14.27.29110
  and Visual Studio  Version: 15.9.28  VS Tools Version: 14.16.27023

**To Reproduce**
`set VCPKG_CRT_LINKAGE=dynamic`
`set VCPKG_DEFAULT_TRIPLET=x64-windows`
`set VCPKG_DISABLE_METRICS=true`
`set VCPKG_DOWNLOADS=[vcpkg-root]\test\downloads`
`set VCPKG_FEATURE_FLAGS=-binarycaching,-compilertracking,-manifests,-registries`
`set VCPKG_LIBRARY_LINKAGE=dynamic`
`set VCPKG_MAX_CONCURRENCY=8`
`set VCPKG_NO_CLEAN=YES`
`set VCPKG_PLATFORM_TOOLSET=[VS Native Toolset for 2017 or 2019 - either v141 or v142`
`set VCPKG_VISUAL_STUDIO_PATH=C:\Program Files (x86)\Microsoft Visual Studio\[VS Version - either 2017 or 2019]\Community`

`git pull` current as of ~18:00 GMT ~14:00 Local
`.\bootstrap-vcpkg.bat -verbose -disableMetrics -win64 -withVSPath "'%VCPKG_VISUAL_STUDIO_PATH%'"`

Run custom script to remove/install all ports found by `.\vcpkg upgrade` as upgrade option does not support --editable and --x-use-aria2 options reference issues #12643 and #12834
log show spdlog failed during install

Try and rerun install of port spdlog with benchmark feature.
`vcpkg install spdlog[benchmark,core]:x64-windows --editable`

**Failure logs** 
- Cut and past the appropriate build messages from the console output.
`Computing installation plan...`
`The following packages will be built and installed:`
`    spdlog[benchmark,core]:x64-windows`
`Starting package 1/1: spdlog:x64-windows`
`Building package spdlog[benchmark,core]:x64-windows...`
`-- Using cached D:/vcpkg_x64/vs2019/prod/downloads/gabime-spdlog-4a9ccf7e38e257feecce0c579a782741254eaeef.tar.gz`
`-- Extracting source D:/vcpkg_x64/vs2019/prod/downloads/gabime-spdlog-4a9ccf7e38e257feecce0c579a782741254eaeef.tar.gz`
`-- Using source at D:/vcpkg_x64/vs2019/prod/buildtrees/spdlog/src/41254eaeef-2548ad24d3`
`-- Configuring x64-windows`
`-- Building x64-windows-dbg`
`CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:142 (message):`
`    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j8`
`    Working Directory: D:/vcpkg_x64/vs2019/prod/buildtrees/spdlog/x64-windows-dbg`
`    See logs for more information:`
`      D:\vcpkg_x64\vs2019\prod\buildtrees\spdlog\install-x64-windows-dbg-out.log`
` `
`Call Stack (most recent call first):`
`  scripts/cmake/vcpkg_build_cmake.cmake:93 (vcpkg_execute_build_process)`
`  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)`
`  ports/spdlog/portfile.cmake:22 (vcpkg_install_cmake)`
`  scripts/ports.cmake:79 (include)`
` `
`Error: Building package spdlog:x64-windows failed with: BUILD_FAILED`
`Please ensure you're using the latest portfiles with `.\vcpkg update`, then`
`submit an issue at https://github.com/Microsoft/vcpkg/issues including:`
`  Package: spdlog:x64-windows`
`  Vcpkg version: 2020.06.15-nohash`
` `
`Additionally, attach any relevant sections from the log files above.`

- Please attach any additional failure logs mentioned in the console output.
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/5447199/install-x64-windows-dbg-out.log)

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.
- Appears to be related to recent upgrade to port fmt https://github.com/microsoft/vcpkg/pull/14239
- Also happens using VS2017 also: 
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/5447220/install-x64-windows-dbg-out.log)

**Other things tried**
I removed spdlog dependencies (ports benchmark and fmt), deleted directories .\buildtrees\benchmark, fmt and spdlog and then tried to reinstall all three ports from scratch with same result... benchmark and fmt installed, spdlog failed with same errors.

`SYSTEM@ANDROMEDA 10/27/2020 14:04:53 D:\vcpkg_x64\vs2019\prod`
`->vcpkg remove fmt`
`The following packages will be removed:`
`    fmt:x64-windows`
`Removing package fmt:x64-windows...`
`Removing package fmt:x64-windows... done`

`SYSTEM@ANDROMEDA 10/27/2020 14:04:53 D:\vcpkg_x64\vs2019\prod`
`->vcpkg remove benchmark`
`The following packages will be removed:`
`    benchmark:x64-windows`
`Removing package benchmark:x64-windows...`
`Removing package benchmark:x64-windows... done`

`SYSTEM@ANDROMEDA 10/27/2020 14:04:53 D:\vcpkg_x64\vs2019\prod`
`->rmdir /q /s .\buildtrees\fmt`

`SYSTEM@ANDROMEDA 10/27/2020 14:05:16 D:\vcpkg_x64\vs2019\prod`
`->rmdir /q /s .\buildtrees\benchmark`

`SYSTEM@ANDROMEDA 10/27/2020 14:05:22 D:\vcpkg_x64\vs2019\prod`
`->rmdir /q /s .\buildtrees\spdlog`

`SYSTEM@ANDROMEDA 10/27/2020 14:05:27 D:\vcpkg_x64\vs2019\prod`
`->vcpkg install spdlog[benchmark,core] fmt[core] benchmark[core] --editable`
`Computing installation plan...`
`The following packages will be built and installed:`
`    benchmark[core]:x64-windows`
`    fmt[core]:x64-windows`
`    spdlog[benchmark,core]:x64-windows`
`Starting package 1/3: benchmark:x64-windows`
`Building package benchmark[core]:x64-windows...`
`-- Note: benchmark only supports static library linkage. Building static library.`
`-- Using cached D:/vcpkg_x64/vs2019/prod/downloads/google-benchmark-73d4d5e8d6d449fc8663765a42aa8aeeee844489.tar.gz`
`-- Extracting source D:/vcpkg_x64/vs2019/prod/downloads/google-benchmark-73d4d5e8d6d449fc8663765a42aa8aeeee844489.tar.gz`
`-- Using source at D:/vcpkg_x64/vs2019/prod/buildtrees/benchmark/src/eeee844489-75b6880063`
`-- Configuring x64-windows`
`-- Building x64-windows-dbg`
`-- Building x64-windows-rel`
`-- Using msys root at D:/vcpkg_x64/vs2019/prod/downloads/tools/msys2/17e91aa90d284835`
`-- Fixing pkgconfig file: D:/vcpkg_x64/vs2019/prod/packages/benchmark_x64-windows/lib/pkgconfig/benchmark.pc`
`-- Fixing pkgconfig file: D:/vcpkg_x64/vs2019/prod/packages/benchmark_x64-windows/debug/lib/pkgconfig/benchmark.pc`
`-- Installing: D:/vcpkg_x64/vs2019/prod/packages/benchmark_x64-windows/share/benchmark/copyright`
`-- Performing post-build validation`
`-- Performing post-build validation done`
`Building package benchmark[core]:x64-windows... done`
`Installing package benchmark[core]:x64-windows...`
`Installing package benchmark[core]:x64-windows... done`
`Elapsed time for package benchmark:x64-windows: 9.658 s`
`Starting package 2/3: fmt:x64-windows`
`Building package fmt[core]:x64-windows...`
`-- Using cached D:/vcpkg_x64/vs2019/prod/downloads/fmtlib-fmt-4fe0b11195b7cd71f39253c44db2c9dddf6b82d4.tar.gz`
`-- Extracting source D:/vcpkg_x64/vs2019/prod/downloads/fmtlib-fmt-4fe0b11195b7cd71f39253c44db2c9dddf6b82d4.tar.gz`
`-- Applying patch fix-warning4189.patch`
`-- Using source at D:/vcpkg_x64/vs2019/prod/buildtrees/fmt/src/dddf6b82d4-f2c5fef865`
`-- Configuring x64-windows`
`-- Building x64-windows-dbg`
`-- Building x64-windows-rel`
`-- Installing: D:/vcpkg_x64/vs2019/prod/packages/fmt_x64-windows/share/fmt/copyright`
`-- Using msys root at D:/vcpkg_x64/vs2019/prod/downloads/tools/msys2/17e91aa90d284835`
`-- Fixing pkgconfig file: D:/vcpkg_x64/vs2019/prod/packages/fmt_x64-windows/lib/pkgconfig/fmt.pc`
`-- Fixing pkgconfig file: D:/vcpkg_x64/vs2019/prod/packages/fmt_x64-windows/debug/lib/pkgconfig/fmt.pc`
`-- Installing: D:/vcpkg_x64/vs2019/prod/packages/fmt_x64-windows/share/fmt/usage`
`-- Performing post-build validation`
`-- Performing post-build validation done`
`Building package fmt[core]:x64-windows... done`
`Installing package fmt[core]:x64-windows...`
`Installing package fmt[core]:x64-windows... done`
`Elapsed time for package fmt:x64-windows: 5.134 s`
`Starting package 3/3: spdlog:x64-windows`
`Building package spdlog[benchmark,core]:x64-windows...`
`-- Using cached D:/vcpkg_x64/vs2019/prod/downloads/gabime-spdlog-4a9ccf7e38e257feecce0c579a782741254eaeef.tar.gz`
`-- Extracting source D:/vcpkg_x64/vs2019/prod/downloads/gabime-spdlog-4a9ccf7e38e257feecce0c579a782741254eaeef.tar.gz`
`-- Using source at D:/vcpkg_x64/vs2019/prod/buildtrees/spdlog/src/41254eaeef-2548ad24d3`
`-- Configuring x64-windows`
`-- Building x64-windows-dbg`
`CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:142 (message):`
`    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j8`
`    Working Directory: D:/vcpkg_x64/vs2019/prod/buildtrees/spdlog/x64-windows-dbg`
`    See logs for more information:`
`      D:\vcpkg_x64\vs2019\prod\buildtrees\spdlog\install-x64-windows-dbg-out.log`
` `
`Call Stack (most recent call first):`
`  scripts/cmake/vcpkg_build_cmake.cmake:93 (vcpkg_execute_build_process)`
`  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)`
`  ports/spdlog/portfile.cmake:22 (vcpkg_install_cmake)`
`  scripts/ports.cmake:79 (include)`
` `
`Error: Building package spdlog:x64-windows failed with: BUILD_FAILED`
`Please ensure you're using the latest portfiles with `.\vcpkg update`, then`
`submit an issue at https://github.com/Microsoft/vcpkg/issues including:`
`  Package: spdlog:x64-windows`
`  Vcpkg version: 2020.06.15-nohash`
` `
`Additionally, attach any relevant sections from the log files above.`

--- a/ports/spdlog/CONTROL
+++ b/ports/spdlog/CONTROL
@@ -1,5 +1,6 @@
 Source: spdlog
 Version: 1.8.0
+Port-Version: 1
 Homepage: https://github.com/gabime/spdlog
 Description: Very fast, header only, C++ logging library
 Build-Depends: fmt

--- a/ports/spdlog/fix-featurebuild.patch
+++ b/ports/spdlog/fix-featurebuild.patch
@@ -1,0 +1,39 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 3a7663b..3e554d6 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -171,7 +171,7 @@ target_link_libraries(spdlog_header_only INTERFACE Threads::Threads)
+ # ---------------------------------------------------------------------------------------
+ if (SPDLOG_FMT_EXTERNAL OR SPDLOG_FMT_EXTERNAL_HO)
+     if (NOT TARGET fmt::fmt)
+-        find_package(fmt 5.3.0 REQUIRED)
++        find_package(fmt CONFIG REQUIRED)
+     endif ()
+     target_compile_definitions(spdlog PUBLIC SPDLOG_FMT_EXTERNAL)
+     target_compile_definitions(spdlog_header_only INTERFACE SPDLOG_FMT_EXTERNAL)
+diff --git a/bench/async_bench.cpp b/bench/async_bench.cpp
+index d4167a4..ccb5406 100644
+--- a/bench/async_bench.cpp
++++ b/bench/async_bench.cpp
+@@ -9,7 +9,7 @@
+ #include "spdlog/spdlog.h"
+ #include "spdlog/async.h"
+ #include "spdlog/sinks/basic_file_sink.h"
+-#include "spdlog/fmt/bundled/locale.h"
++#include <fmt/locale.h>
+ 
+ #include "utils.h"
+ #include <atomic>
+diff --git a/bench/bench.cpp b/bench/bench.cpp
+index b7d2fc7..e2f8109 100644
+--- a/bench/bench.cpp
++++ b/bench/bench.cpp
+@@ -11,7 +11,7 @@
+ #include "spdlog/sinks/daily_file_sink.h"
+ #include "spdlog/sinks/null_sink.h"
+ #include "spdlog/sinks/rotating_file_sink.h"
+-#include "spdlog/fmt/bundled/locale.h"
++#include <fmt/locale.h>
+ 
+ #include "utils.h"
+ #include <atomic>

--- a/ports/spdlog/portfile.cmake
+++ b/ports/spdlog/portfile.cmake
@@ -4,6 +4,7 @@ vcpkg_from_github(
     REF 4a9ccf7e38e257feecce0c579a782741254eaeef # v1.8.0
     SHA512 333f14704e0d0aa88abbe4ddd29aeb009de2f845440559d463f1b7f9c7da32b2fbdba0f2abf97ec2a5c479d2d62bb2220b21a1bc423d62fbbb93952cf829d532
     HEAD_REF v1.x
+    PATCHES fix-featurebuild.patch
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS

